test/GPv2TradeExecution.test.ts

```diff
@@ -144,4 +144,107 @@ describe("GPv2TradeExecution", () => {
       });
     });
   });
+
+  describe("transferBuyAmountToOwner", () => {
+    const withoutSell = {
+      sellToken: ethers.constants.AddressZero,
+      sellAmount: ethers.constants.Zero,
+    };
+
+    it("should transfer buy amount to sender", async () => {
+      const amount = ethers.utils.parseEther("13.37");
+
+      const buyToken = await waffle.deployMockContract(deployer, IERC20.abi);
+      await buyToken.mock.transfer
+        .withArgs(traders[0].address, amount)
+        .returns(true);
+
+      await expect(
+        tradeExecution.transferBuyAmountToOwnerTest(
+          composeExecutedTrade({
+            owner: traders[0].address,
+            buyToken: buyToken.address,
+            buyAmount: amount,
+            ...withoutSell,
+          }),
+        ),
+      ).to.not.be.reverted;
+    });
+
+    it("should revert on failed ERC20 transfers", async () => {
+      const amount = ethers.utils.parseEther("4.2");
+
+      const buyToken = await waffle.deployMockContract(deployer, IERC20.abi);
+      await buyToken.mock.transfer
+        .withArgs(traders[0].address, amount)
+        .revertsWithReason("test error");
+
+      await expect(
+        tradeExecution.transferBuyAmountToOwnerTest(
+          composeExecutedTrade({
+            owner: traders[0].address,
+            buyToken: buyToken.address,
+            buyAmount: amount,
+            ...withoutSell,
+          }),
+        ),
+      ).to.be.revertedWith("test error");
+    });
+
+    it("should revert when transfering from an address without code", async () => {
```

Comment:
If my suggestion in the other PR is accepted, also this should change:
```suggestion
    it("should revert when transfering from a token with no contract at its address", async () => {
```